checkov/common/runners/base_runner.py

```diff
@@ -12,11 +12,36 @@
 
 class BaseRunner(ABC):
     check_type = ""
+    definitions = None
+    context = None
+    breadcrumbs = None
+    external_registries = None
+    graph_manager = None
+    graph_registry = None
 
     @abstractmethod
     def run(self, root_folder, external_checks_dir=None, files=None, runner_filter=RunnerFilter(), collect_skip_comments=True):
         pass
 
+    def set_external_data(self, definitions: Optional[Dict], context: Optional[Dict], breadcrumbs: Optional[Dict]):
+        self.definitions = definitions
+        self.context = context
+        self.breadcrumbs = breadcrumbs
+
+    def load_external_checks(self, external_checks_dir: List[str]):
+        pass
+
+    def get_graph_checks_report(self, root_folder, runner_filter: RunnerFilter):
+        pass
+
+    def run_graph_checks_results(self, runner_filter):
+        checks_results = {}
+        for r in self.external_registries + [self.graph_registry]:
```

Comment:
this can be written without creating an extra list
```suggestion
        for r in itertools.chain(self.external_registries, [self.graph_registry]):
```